src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -772,6 +773,7 @@ private PackageDependency CreateDependencyFromProject(dynamic project, Dictionar
                         string.Format(CultureInfo.CurrentCulture, NuGetResources.ProjectJsonPack_Deprecated, builder.Id),
                         NuGetLogCode.NU5126));
                 }
+#pragma warning restore CS0612 // Type or member is obsolete
```

Comment:
What type or member is marked as obsolete? Is it `ProcessJsonFile`?  This `#pragma warning restore` can be moved to line 766, so that the rest of the code still gets checked by the analyzer.

Reply:
Yes, the `ProcessJsonFile` method is obsolete. I applied the Roslyn analyzer suggestion, which applied the suppression to the entire if-else block. Given that you have approved this PR, and this [if-else block has not been touched in the last 5 years](https://github.com/NuGet/NuGet.Client/blame/dev/src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs#L764-L774), I am leaning towards merging this pull request.

Reply:
I'm not sure why the last modified date of the block is interesting in making this decision.

I don't imagine that `Logger.Log` or `PackagingLogMessage.CreateWarning` will ever become marked as obsolete. However, I can imagine that there's a possibility that maybe `projectFactory.ProcessNuspec` gets marked as obsolete one day. With this PR the way it is, if `ProcessNuspec` does get marked obsolete in the future, this usage will not get warned.

Reply:
> I'm not sure why the last modified date of the block is interesting in making this decision.

My thoughts were about the necessity of running the CI by altering the location for restoring the warning. I am happy to submit a follow-up PR, ensuring that team members won’t have to be concerned with local build failures if they happen to run the builds on the latest dogfooding versions of the tooling.

Reply:
Created follow-up PR addressing this feedback https://github.com/NuGet/NuGet.Client/pull/5715.